recognition/include/pcl/recognition/impl/hv/hv_papazov.hpp

```diff
@@ -141,12 +141,12 @@ template<typename ModelT, typename SceneT>
       typename boost::graph_traits<Graph>::adjacency_iterator ai;
       typename boost::graph_traits<Graph>::adjacency_iterator ai_end;
 
-      boost::shared_ptr < RecognitionModel > current = static_cast<boost::shared_ptr<RecognitionModel> > (graph_id_model_map_[int (v)]);
+      auto current = boost::dynamic_pointer_cast<RecognitionModel> (graph_id_model_map_[int (v)]);
```

Comment:
Is it worth to incur into the dynamic cast runtime penalty here? The subsequent logic is not handling null returns.

Reply:
Right, I will replace this with `boost::static_pointer_cast`.

Reply:
I will squash this fix in corresponding commit right away.